### PR TITLE
Fix daml ledger navigator configuration.

### PR DESCRIPTION
**Pull request by associahedron**
Bugfix: `daml ledger navigator` was not configuring the navigator properly -- it was surreptitiously using the daml.yaml from the DAML_PROJECT directory instead of the configuration we wrote, and that configuration was itself not correct. Sadly I can't easily write a regression test for this...

I'm still not certain everything is right for all ledgers, but it works for Sandbox, and I'm sure we'll root out any remaining issues quickly, with Canton testing the navigator. (Note, this is still a workaround for #2418)

### CODE REVIEW

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs [hunk 1]
```diff
@@ -795,35 +795,46 @@ runLedgerNavigator flags remainingArguments = do
...
 
   where
     navigatorConfig :: [PartyDetails] -> T.Text
-    navigatorConfig partyDetails =
-        TL.toStrict . encodeToLazyText $ object
-            ["users" .= object
-                [ TL.toStrict displayName .= object [ "party" .= TL.toStrict (unParty party) ]
-                | PartyDetails{..} <- partyDetails
-                ]
+    navigatorConfig partyDetails = T.unlines . concat $
```
Comment (cocreature) [inline daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs]: Why did you change the format here?

Comment (associahedron) [reply daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs]: I *think* the json format doesn't work, but I could be wrong. This might have just been the case of fixing the wrong thing by accident, before fixing the right thing.

But this format definitely works.

Comment (cocreature) [reply daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs]: It definitely used to work at some point and iirc that was a feature of the underlying configuration lib so I would be surprised if it has changed. Using a standard format such as JSON does seem nicer than using whatever weird format that scala config lib is using but given that this will go away with #2418, it’s not worth spending any effort trying to figure out what’s going wrong :slightly_smiling_face:

Comment (associahedron) [reply daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs]: I agree with everything you say :)
